src/subscription/Subscription.js

```diff
@@ -79,19 +79,44 @@ Subscription.prototype.subscribed = function() {
  * @return {boolean}
  */
 Subscription.prototype.alive = function() {
-    return this.subscribed() && Date.now() < this.expirationTime();
+    return this.subscribed() && Date.now() < (this.expirationTime() - this._renewHandicapMs);
 };
 
 /**
  * @return {boolean}
  */
 Subscription.prototype.expired = function() {
     if (!this.subscribed()) return true;
-    return !this.subscribed() || Date.now() > this.subscription().expirationTime;
+    return !this.subscribed() || Date.now() > this.expirationTime();
```

Comment:
You should create a helper function `_parseExpirationTime` instead of this and keep the original logic. We don't know who and how relies on it, so the smaller the change the better.